### PR TITLE
Solver options

### PR DESCRIPTION
**Pull request by MaximeBouton**
This PR addresses #9 

- `create_policy` no longer exists
- `ValueIterationPolicy` has outer constructors rather than inner ones. 
- the solve method is `solve(solver, mdp)`
- the solver type has new fields: `verbose`, `include_Q` and `init_util`

### CODE REVIEW

File: src/vanilla.jl [hunk 1]
```diff
@@ -16,60 +23,51 @@ mutable struct ValueIterationPolicy <: Policy
...
-        self.policy = policy
-        self.action_map = ordered_actions(mdp)
-        self.include_Q = true
-        self.mdp = mdp
-        return self
+end
+
+# constructor with an optinal initial value function argument
+function ValueIterationPolicy(mdp::Union{MDP,POMDP};
+                              utility::Vector{Float64}=Vector{Float64}(0),
```
Comment (rejuvyesh) [inline src/vanilla.jl]: Why not:

```julia
utility::Vector{Float64}=zeros(n_states(mdp)),
policy::Vector{Int64}=zeros(Int64, n_states(mdp)),
```

Then you won't need the extra if else.

Comment (MaximeBouton) [reply src/vanilla.jl]: Good point, I made the change.

**Review by zsunberg — APPROVED** [review]
Sorry for the delay - I was travelling, but thanks so much for doing this!!! This looks great! Switching to outer constructors makes my heart happy, haha! :heart: :heart:

**Comment by zsunberg** [comment]
Just thought of this - test QMDP against this before you merge it!!!

**Comment by MaximeBouton** [comment]
I am preparing a PR on QMDP. I will merge this one after

**Comment by MaximeBouton** [comment]
JuliaPOMDP/QMDP.jl#10

**Pull request merged**